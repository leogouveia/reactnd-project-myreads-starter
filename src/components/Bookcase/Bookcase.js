import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { Shelf } from "../Shelf/Shelf";
import { Book } from "../Book/Book";
import { ShelfChange } from "../ShelfChange/ShelfChange";

class Bookcase extends React.Component {
  static propTypes = {
    shelves: PropTypes.array,
    books: PropTypes.array,
    handleChangeShelf: PropTypes.func
  };

  render() {
    const { shelves, books, handleChangeShelf } = this.props;
    return (
      <div>
        {shelves.map(bookshelf => (
          <Shelf key={bookshelf.id} name={bookshelf.name}>
            {books.filter(book => book.shelf === bookshelf.id).map(book => (
              <li key={book.id}>
                <Book book={book}>
                  <ShelfChange
                    book={book}
                    shelves={shelves}
                    handleChangeShelf={handleChangeShelf}
                  />
                </Book>
              </li>
            ))}
          </Shelf>
        ))}
        <div className="open-search">
          <Link to="/search">Add book</Link>
        </div>
      </div>
    );
  }
  
}

export default Bookcase;
