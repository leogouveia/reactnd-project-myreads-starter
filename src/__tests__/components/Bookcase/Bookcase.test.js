import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Bookcase from "../../../components/Bookcase/Bookcase";
import { shelves, books } from "../../fixtures/fixtures";

describe("<Bookcase />", () => {
  it("should render a Bookcase", () => {
    const handleChangeShelf = jest.fn();
    const wrapper = shallow(
      <Bookcase
        shelves={shelves}
        books={books}
        handleChangeShelf={handleChangeShelf}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
