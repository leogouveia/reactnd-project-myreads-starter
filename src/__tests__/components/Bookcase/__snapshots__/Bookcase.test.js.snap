// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bookcase /> should render a Bookcase 1`] = `
<div>
  <Shelf
    key="shelf1"
    name="Shelf 1"
  >
    <li
      key="1"
    >
      <Book
        authors={Array []}
        book={
          Object {
            "authors": "Jonh Doe",
            "id": 1,
            "imageLinks": Object {
              "thumbnail": "http://fakeimage.com/image1.png",
            },
            "shelf": "shelf1",
            "title": "Book1 Title",
          }
        }
      >
        <ShelfChange
          book={
            Object {
              "authors": "Jonh Doe",
              "id": 1,
              "imageLinks": Object {
                "thumbnail": "http://fakeimage.com/image1.png",
              },
              "shelf": "shelf1",
              "title": "Book1 Title",
            }
          }
          handleChangeShelf={[MockFunction]}
          shelves={
            Array [
              Object {
                "id": "shelf1",
                "name": "Shelf 1",
              },
              Object {
                "id": "shelf2",
                "name": "Shelf 2",
              },
              Object {
                "id": "shelf3",
                "name": "Shelf 3",
              },
            ]
          }
        />
      </Book>
    </li>
  </Shelf>
  <Shelf
    key="shelf2"
    name="Shelf 2"
  />
  <Shelf
    key="shelf3"
    name="Shelf 3"
  >
    <li
      key="2"
    >
      <Book
        authors={Array []}
        book={
          Object {
            "authors": "Jane Doe",
            "id": 2,
            "imageLinks": Object {
              "thumbnail": "http://fakeimage.com/image2.png",
            },
            "shelf": "shelf3",
            "title": "Book2 Title",
          }
        }
      >
        <ShelfChange
          book={
            Object {
              "authors": "Jane Doe",
              "id": 2,
              "imageLinks": Object {
                "thumbnail": "http://fakeimage.com/image2.png",
              },
              "shelf": "shelf3",
              "title": "Book2 Title",
            }
          }
          handleChangeShelf={[MockFunction]}
          shelves={
            Array [
              Object {
                "id": "shelf1",
                "name": "Shelf 1",
              },
              Object {
                "id": "shelf2",
                "name": "Shelf 2",
              },
              Object {
                "id": "shelf3",
                "name": "Shelf 3",
              },
            ]
          }
        />
      </Book>
    </li>
  </Shelf>
  <div
    className="open-search"
  >
    <Link
      replace={false}
      to="/search"
    >
      Add book
    </Link>
  </div>
</div>
`;
